server/aws-lsp-codewhisperer/src/language-server/agenticChat/agenticChatController.ts

```diff
@@ -135,6 +135,8 @@ export class AgenticChatController implements ChatHandlers {
     #additionalContextProvider: AdditionalContextProvider
     #contextCommandsProvider: ContextCommandsProvider
     #stoppedToolUses = new Set<string>()
+    // Map to store approved paths per tab to avoid repeated validation
+    #approvedPaths = new Map<string, Set<string>>()
```

Comment:
Is it easy to seperate this permissions logic into a new component. Something like `AgenticChatPermissions`? I think eventually we will want to build on this work in the future with more complex options (Like trusting tools). 

Splitting it into its own component will also make it much easier to test.